src/main/java/picard/sam/util/ReadNameParser.java

```diff
@@ -29,7 +32,7 @@
      */
     public static final String DEFAULT_READ_NAME_REGEX = "<optimized capture of last three ':' separated fields as numeric values>".intern();
 
-    private Map<String, PhysicalLocation> locationMap = new HashMap<>();
+    private Cache<String, PhysicalLocation> locationCache = CacheBuilder.newBuilder().build();
```

Comment:
without a `maximumSize` this doesn't actually evict any entries....

Reply:
what should I set as maximumSize?

Reply:
I dont' think it should matter much. I imagine that 1 would be enough....but we can use 20 to be "safe"